### PR TITLE
Annotations refactor

### PR DESCRIPTION
**Pull request by cmsigrist**
Remove all the @RequireAndroid O annotations and most of the Android N (some remain in the audio directory, for the chronometers to work)

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileBirthday.kt [hunk 1]
```diff
@@ -70,12 +71,19 @@ class ProfileBirthday : AppCompatActivity() {
...
-        ).years
+    private fun getAge(year: Int, month: Int, day: Int): Int {
+        val calendar = GregorianCalendar()
+
+        val y = calendar.get(Calendar.YEAR)
+        //For some unknown reason, months are indexed from 0 to 11...
+        val m = calendar.get(Calendar.MONTH) + 1
+        val d = calendar.get(Calendar.DAY_OF_MONTH)
+        var age = y - year
+        if ((m < month) || ((m == month) && (d < day))) {
```
Comment (koko-ng) [inline app/src/main/java/ch/epfl/sdp/blindly/profile_setup/ProfileBirthday.kt:82]: Maybe comment this lines to explain what they do, it's a bit cryptic to me

File: app/src/main/java/ch/epfl/sdp/blindly/user/User.kt [hunk 2]
```diff
@@ -298,17 +298,22 @@ class User private constructor(
...
-            return Period.between(
-                LocalDate.of(
-                    year.toInt(),
-                    month.toInt(),
-                    day.toInt()
-                ),
-                LocalDate.now()
-            ).years
+            val (day, month, year) = birthday.split('.').map { s -> s.toInt() }
+            val calendar = GregorianCalendar()
```
Comment (koko-ng) [inline app/src/main/java/ch/epfl/sdp/blindly/user/User.kt:303]: This snippet is in two different places, maybe a date utils class could be helpful?

**Review by koko-ng — COMMENTED** [review]
Nice work, thanks!

**Review by koko-ng — APPROVED** [review]

File: .idea/codeStyles/Project.xml [hunk 3]
```diff
@@ -1,22 +1,6 @@
...
-      </option>
-      <option name="PACKAGES_IMPORT_LAYOUT">
-        <value>
-          <package name="" alias="false" withSubpackages="true" />
-          <package name="java" alias="false" withSubpackages="true" />
-          <package name="javax" alias="false" withSubpackages="true" />
-          <package name="kotlin" alias="false" withSubpackages="true" />
-          <package name="" alias="true" withSubpackages="true" />
-        </value>
-      </option>
```
Comment (Lollipoke) [inline .idea/codeStyles/Project.xml:19]: I think this happens when you don't run connectedCheck, maybe we should keep the previous version ?

File: app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/ProfileBirthdayTest.kt [hunk 4]
```diff
@@ -25,7 +25,7 @@ import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 
-const val TEST_BIRTHDAY = "18.03.2003"
+const val TEST_BIRTHDAY = "16.05.2003"
```
Comment (Lollipoke) [inline app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/ProfileBirthdayTest.kt:28]: What's the point of changing the date ? Both are > 18 no ?

File: app/src/main/java/ch/epfl/sdp/blindly/main_screen/match/cards/CardStackAdapter.kt [hunk 5]
```diff
@@ -48,7 +48,7 @@ class CardStackAdapter(
         val profile = profiles[position]
         val v = holder.itemView.findViewById(R.id.item_image) as ImageView
         v.setImageResource(R.drawable.background)
-        holder.name_age.text = "${profile.name}, ${profile.age}"
+        holder.nameAge.text = "${profile.name}, ${profile.age}"
```
Comment (Lollipoke) [inline app/src/main/java/ch/epfl/sdp/blindly/main_screen/match/cards/CardStackAdapter.kt:51]: I fixed that in my PR ;)

File: app/src/main/java/ch/epfl/sdp/blindly/user/User.kt (hunk 2 above)
Comment (Lollipoke) [reply app/src/main/java/ch/epfl/sdp/blindly/user/User.kt:303]: I don't know if a class for one method is really necessary, but maybe a public method could be a good idea indeed !

**Review by Lollipoke — APPROVED** [review]
Great job !

File: app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/ProfileBirthdayTest.kt (hunk 4 above)
Comment (cmsigrist) [reply app/src/androidTest/java/ch/epfl/sdp/blindly/profile_setup/ProfileBirthdayTest.kt:28]: Yes, indeed, but it was to test the corner case of m == month and d <= day in getUserAge()

File: .idea/codeStyles/Project.xml (hunk 3 above)
Comment (cmsigrist) [reply .idea/codeStyles/Project.xml:19]: Ohhhh! That might be it, thanks (it might have come with the update of Android Studio)!

Comment (cgntnr) [reply .idea/codeStyles/Project.xml:19]: We can simply not add these to the commit anyways :D Would be an easier solution I believe.

Comment (cmsigrist) [reply .idea/codeStyles/Project.xml:19]: Yep I'll exclude it :D
